app/components/post_draft/post_draft.js

```diff
@@ -59,6 +61,9 @@ export default class PostDraft extends PureComponent {
         userIsOutOfOffice: PropTypes.bool.isRequired,
         value: PropTypes.string.isRequired,
         valueEvent: PropTypes.string,
+        useGroupMentions: PropTypes.bool.isRequired,
+        channelMemberCountsByGroup: PropTypes.object,
+        groupsWithAllowReference: PropTypes.object,
```

Comment:
Do these needs to be objects? as I remember in the reducer we were returning a new state every time and by receiving an object as a prop the component will surely re-render every time they change, is there a way to avoid that? If not why not?

Reply:
Yes. I was able to stop unnecessary re-renders for channelMemberCountsByGroup by doing a deep equal check on it's properties. If the properties don't match or the entire object does not exist in the state, we want to return a new state, otherwise, if the current groups all exist in the state and their properties are the same, we can return current state which will avoid re-rendering.

Reply:
As for the groupsWithAllowReference, it's an object that's converted to a Map but not actually an object. I just couldn't find `PropTypes.map` so I used object instead. I'm not too sure how to handle this one since it's existing reducer?

Reply:
> I was able to stop unnecessary re-renders for channelMemberCountsByGroup by doing a deep equal check on it's properties

where is this happening?

Reply:
@enahum 
We can see that if the group doesn't have any group members in it, the action doesn't get dispatched.
If it does, the action `RECEIVED_CHANNEL_MEMBER_COUNT_BY_GROUP` gets dispatched which now has more of a detailed check wether or not to return new state or not.
![image](https://user-images.githubusercontent.com/17804942/87479155-f919cb80-c5f8-11ea-9a04-a76acfc062b4.png)
![image](https://user-images.githubusercontent.com/17804942/87479164-fcad5280-c5f8-11ea-80fb-61e6c1b02b46.png)